tests/state/test_helpers.py

```diff
@@ -0,0 +1,169 @@
+import pytest
+
+from beacon_chain.state.config import (
+    generate_config,
+)
+from beacon_chain.state.state_transition import (
+    get_shuffling,
+)
+from beacon_chain.state.validator_record import (
+    ValidatorRecord,
+)
+from beacon_chain.state.helpers import (
+    get_crosslink_shards,
+    get_crosslink_notaries,
+    get_crosslink_shards_count,
+)
+
+
+# Fixed configuration for this test especially
+REPLACED_PARAMETERS = {
+    'shard_count': 20,
+    'notaries_per_crosslink': 100,
+}
+testing_helpers_config = generate_config(
+    shard_count=20,
+    notaries_per_crosslink=100,
+)
+
+
+def create_crystallized_state(
+        genesis_crystallized_state,
+        init_shuffling_seed, next_shard,
+        active_validators=None):
+    crystallized_state = genesis_crystallized_state
+    crystallized_state.next_shard = next_shard
+    if active_validators is not None:
+        crystallized_state.active_validators = active_validators
+        crystallized_state.current_shuffling = get_shuffling(
+            init_shuffling_seed,
+            len(active_validators),
+        )
+    return crystallized_state
+
+
+def mock_validator_record(pubkey, testing_helpers_config):
+    return ValidatorRecord(
+        pubkey=pubkey,
+        withdrawal_shard=0,
+        withdrawal_address=pubkey.to_bytes(32, 'big')[-20:],
+        randao_commitment=b'\x55'*32,
+        balance=testing_helpers_config['default_balance'],
+        switch_dynasty=testing_helpers_config['default_switch_dynasty']
+    )
+
+
+@pytest.mark.parametrize(
+    'active_validators_count, expected',
+    (
+        (1000, 10),
+        (10000, testing_helpers_config['shard_count'])
+    )
+)
+def test_get_crosslink_shards_count(active_validators_count, expected):
```

Comment:
I prefer to use config fixtures like this. By overriding `shard_count` and `notaries_per_crosslink` in the `parametrize` it actually ripples through the `config` fixture that you get.
```python
@pytest.mark.parametrize(
    'active_validators_count,shard_count,notaries_per_crosslink,expected',
    (
        (1000, 20, 100, 10),
        (10000, 20, 100, 20),
        (50, 100, 3, 16),
    )
)
def test_get_crosslink_shards_count(active_validators_count,
                                    shard_count,
                                    notaries_per_crosslink,
                                    config,
                                    expected):
    crosslink_shards_count = get_crosslink_shards_count(
        active_validators_count,
        config=config,
    )
    assert crosslink_shards_count == expected
```
(this hinges upon defining a `notaries_per_crosslink` fixture and having it feed into the `config` fixture like the rest of the config vars)

I find it a little bit more readable, and it also gives you the ability to easily try more varied combinations of configurations in your test. Notice the `(50, 100, 3, 16)` test that changes up the config.